src/test/java/net/snowflake/ingest/FipsCompliantIngestIT.java

```diff
@@ -0,0 +1,328 @@
+import net.snowflake.ingest.connection.IngestStatus;
+import net.snowflake.ingest.example.IngestExampleHelper;
+import net.snowflake.ingest.fips.BouncyCastleFIPSInit;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.spec.RSAKeyGenParameterSpec;
+import java.sql.Connection;
+import java.time.Instant;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * This test is specifically testing REST apis with FIPS compliance mode
+ * enabled.
+ * <p>
+ * <b>Prerequisites:</b>
+ * You must have a valid set of Snowflake credentials
+ * We use encrypted key pair so follow the instructions written on the
+ * documentation and also in readme of this repo.
+ */
+public class FipsCompliantIngestIT
+{
+  // Details required to connect to Snowflake over jdbc and establish
+  // pre-requesites
+  private static final String ACCOUNT = "s3testaccount";
+  private static final String USER = "username";
+  private static final String HOST = "s3testaccount.snowflake.com";
+  private static final String SCHEME = "http";
+  private static final String PASSWORD = "***";
+  private static final String ALGORITHM = "RSA";
+  private static final int PORT = 8082;
+
+  // Details for the pipe which we are going to use
+  //the name of our target DB
+  private static final String DATABASE = "testdb";
+  //the name of our target schema
+  private static final String SCHEMA = "public";
+  //the name of our stage
+  private static final String STAGE = "ingest_stage";
+  //the name of our target table
+  private static final String TABLE = "ingest_table";
+  //the name of our pipe
+  private static final String PIPE = "ingest_pipe";
+
+  //the connection we will use for queries
+  private static Connection conn;
+
+  //the Administrative connection
+  //the fully qualified pipe name
+  private static final String FQ_PIPE = DATABASE + "." + SCHEMA + "." + PIPE;
+
+  //the actual ingest manager
+  private static SimpleIngestManager manager;
+
+  //our keypair
+  private static KeyPair keypair;
+
+  // Set of files to ingest
+  private static Set<String> files = new TreeSet<>();
+
+  /**
+   * Creates the stages and files we'll use for this test
+   */
+  private static void setup(Set<String> files, String filesLocation)
+      throws Exception
+  {
+    //use the right database
+    IngestExampleHelper.doQuery(conn, "use database " + DATABASE);
+
+    //use the right schema
+    IngestExampleHelper.doQuery(conn, "use schema " + SCHEMA);
```

Comment:
Why dont you do "create or replace" or "create if not exists"?

Reply:
Yes, much better, will do that!